// ng7-mat-breadcrumb/esm2015/lib/ng7-mat-breadcrumb.module.js
import { NgModule } from '@angular/core';
import { Ng7MatBreadcrumbComponent } from './ng7-mat-breadcrumb.component';
import { RouterModule } from '@angular/router';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatListModule } from '@angular/material/list';
import * as i0 from "@angular/core";
export class Ng7MatBreadcrumbModule {
}
Ng7MatBreadcrumbModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: Ng7MatBreadcrumbModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
Ng7MatBreadcrumbModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: Ng7MatBreadcrumbModule, declarations: [Ng7MatBreadcrumbComponent], imports: [RouterModule,
        CommonModule,
        BrowserAnimationsModule,
        MatToolbarModule,
        MatListModule], exports: [Ng7MatBreadcrumbComponent] });
Ng7MatBreadcrumbModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: Ng7MatBreadcrumbModule, imports: [[
            RouterModule,
            CommonModule,
            BrowserAnimationsModule,
            MatToolbarModule,
            MatListModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: Ng7MatBreadcrumbModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Ng7MatBreadcrumbComponent],
                    imports: [
                        RouterModule,
                        CommonModule,
                        BrowserAnimationsModule,
                        MatToolbarModule,
                        MatListModule
                    ],
                    exports: [Ng7MatBreadcrumbComponent]
                }]
        }] });
