// mat-breadcrumb/esm5/lib/mat-breadcrumb.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
var MatBreadcrumbService = /** @class */ (function () {
    function MatBreadcrumbService() {
        this.breadcrumbLabels = new BehaviorSubject([]);
        this.newBreadcrumb = new BehaviorSubject([]);
    }
    MatBreadcrumbService.prototype.updateBreadcrumbLabels = function (labels) {
        this.breadcrumbLabels.next(labels);
    };
    MatBreadcrumbService.prototype.updateBreadcrumb = function (newBreadcrumb) {
        this.newBreadcrumb.next(newBreadcrumb);
    };
    MatBreadcrumbService.ɵprov = i0.ɵɵdefineInjectable({ factory: function MatBreadcrumbService_Factory() { return new MatBreadcrumbService(); }, token: MatBreadcrumbService, providedIn: "root" });
    MatBreadcrumbService = __decorate([
        Injectable({
            providedIn: 'root'
        })
    ], MatBreadcrumbService);
    return MatBreadcrumbService;
}());
export { MatBreadcrumbService };
